numpy/_core/src/multiarray/stringdtype/dtype.c

```diff
@@ -185,6 +185,41 @@ _eq_comparison(int scoerce, int ocoerce, PyObject *sna, PyObject *ona)
     return na_eq_cmp(sna, ona);
 }
 
+// currently this can only return 1 or -1, the latter indicating that the
+// error indicator is set
+NPY_NO_EXPORT int
+stringdtype_compatible_na(PyObject *na1, PyObject *na2) {
+    if ((na1 == NULL) != (na2 == NULL)) {
```

Comment:
This should be `((na1 == NULL) || (na2 == NULL))`, right? Both `NULL` is definitely fine.